swarm/bmt/bmt.go

```diff
@@ -29,123 +29,111 @@ import (
 /*
 Binary Merkle Tree Hash is a hash function over arbitrary datachunks of limited size
 It is defined as the root hash of the binary merkle tree built over fixed size segments
-of the underlying chunk using any base hash function (e.g keccak 256 SHA3)
+of the underlying chunk using any base hash function (e.g keccak 256 SHA3).
+Chunk with data shorter than the fixed size are hashed as if they had zero padding
 
-It is used as the chunk hash function in swarm which in turn is the basis for the
+BMT hash is used as the chunk hash function in swarm which in turn is the basis for the
 128 branching swarm hash http://swarm-guide.readthedocs.io/en/latest/architecture.html#swarm-hash
 
 The BMT is optimal for providing compact inclusion proofs, i.e. prove that a
 segment is a substring of a chunk starting at a particular offset
-The size of the underlying segments is fixed at 32 bytes (called the resolution
-of the BMT hash), the EVM word size to optimize for on-chain BMT verification
+The size of the underlying segments is fixed to the size of the base hash (called the resolution
+of the BMT hash), Using Keccak256 SHA3 hash is 32 bytes, the EVM word size to optimize for on-chain BMT verification
 as well as the hash size optimal for inclusion proofs in the merkle tree of the swarm hash.
 
 Two implementations are provided:
 
 * RefHasher is optimized for code simplicity and meant as a reference implementation
+  that is simple to understand
 * Hasher is optimized for speed taking advantage of concurrency with minimalistic
   control structure to coordinate the concurrent routines
-  It implements the ChunkHash interface as well as the go standard hash.Hash interface
-
+  It implements the following interfaces
+	* standard golang hash.Hash
+	* SwarmHash
+	* io.Writer
+	* io.ReaderFrom
+	* TODO: SegmentWriter
 */
 
 const (
-	// DefaultSegmentCount is the maximum number of segments of the underlying chunk
-	DefaultSegmentCount = 128 // Should be equal to storage.DefaultBranches
-	// DefaultPoolSize is the maximum number of bmt trees used by the hashers, i.e,
+	// SegmentCount is the maximum number of segments of the underlying chunk
+	// Should be equal to storage.DefaultBranches
+	SegmentCount = 128
+	// PoolSize is the maximum number of bmt trees used by the hashers, i.e,
 	// the maximum number of concurrent BMT hashing operations performed by the same hasher
-	DefaultPoolSize = 8
+	PoolSize = 8
 )
 
-// BaseHasher is a hash.Hash constructor function used for the base hash of the  BMT.
+// BaseHasher is a hash.Hash constructor function used for the base hash of the BMT.
+// implemented by Keccak256 SHA3 sha3.NewKeccak256
 type BaseHasher func() hash.Hash
```

Comment:
Suggestion: I suggest adding `Func` to the end of this name, because we already have a `Hasher` struct, and when seen in code, it is not clear immediately that this is a function.

Reply:
ok